File: 1/rb/calories.rb
def run
    file_data = read_file("../calorie_counter.txt")

    puts get_max_cal(file_data)
    puts get_top_three(file_data)
end

def read_file(path)
    file = File.read(path).split("\n")
end

def get_max_cal(data)
    total, calories = 0, []
    
    data.each do |cal|
        if cal != ""
            total += cal.to_i
        else
            calories.append(total)
            total = 0
        end
    end

    return calories.max()
end

def get_top_three(data)
    total, calories, top_three = 0, [], []

    data.each do |cal|
        if cal != ""
            total += cal.to_i
        else
            calories.append(total)
            total = 0
        end
    end

    for i in 0..2 do
        top_three.append(calories.max())
        puts calories.max()
        calories.delete(calories.max())
    end

    top_three.sum()
end

File: 1/py/calories.py
def run():
    all_cal = read_file('../calorie_counter.txt')
    print(all_cal)

    largest_cal = get_largest_cal(all_cal)
    print(largest_cal)

    top_three = get_top_three(all_cal)
    print(top_three)

def read_file(path):
    output = []

    with open(path) as file:
        for line in file:
            output.append(line.strip())

    return output

def get_largest_cal(all_cal):
    calories = []
    total = 0

    for cal in all_cal:
        if not cal == '':
            total += int(cal)
        else:
            calories.append(total)
            total = 0
    
    return max(calories)

def get_top_three(all_cal):
    calories = []
    top_three = []
    total = 0

    for cal in all_cal:
        if not cal == '':
            total += int(cal)
        else:
            calories.append(total)
            total = 0
    
    for i in range(3):
        top_three.append(max(calories))
        calories.remove(max(calories))

    return sum(top_three)
